event-based-simulators/extras/GenerateMeasurements.py

```diff
@@ -0,0 +1,248 @@
+
+
+def JsonMeasurementList(device_id, distribution):
+    time_pointer = sim_time
+    time_count = 0
+    jsonMeasurementList = []
+    storageMeasurementDict = CreateStorageMeasurementDict()
+    storageMeasurementDict_1800s = CreateStorageMeasurementDict()
+    while time_pointer <= sim_end_time:
+        time_count += 1
+        measurementsTemplateDict = CreateExtraInfoDict(time_pointer, device_id)
+        # 600s
+        for measurementKey in LIST_OF_MEASUREMENTS:
+            newMeasurement = CreateMeasurement(distribution)
+            measurementsDict = CreateIndividualMeasurementDict(measurement_key=measurementKey,
+                                                               measurement_series='600s',
+                                                               new_measurement=newMeasurement)
+
+            measurementsTemplateDict.update(measurementsDict)
+            storageMeasurementDict[f"{measurementKey}"].append(newMeasurement)
+        jsonMeasurementList.append(measurementsTemplateDict)
+
+        # 1800s
+        if time_count == 3 or time_count == 6:
+            measurementsTemplateDict = CreateExtraInfoDict(time_pointer, device_id)
+            for measurementKey in LIST_OF_MEASUREMENTS:
+                newMeasurement = CalculateMeanValue(storage_measurement_dict=storageMeasurementDict,
+                                                    measurement_key=measurementKey)
+                measurementsDict = CreateIndividualMeasurementDict(measurement_key=measurementKey,
+                                                                   measurement_series='1800s',
+                                                                   new_measurement=newMeasurement)
+
+                measurementsTemplateDict.update(measurementsDict)
+                storageMeasurementDict_1800s[f"{measurementKey}"].append(newMeasurement)
+            jsonMeasurementList.append(measurementsTemplateDict)
+
+        # 3600s
+        if time_count == 6:
+            measurementsTemplateDict = CreateExtraInfoDict(time_pointer, device_id)
+            for measurementKey in LIST_OF_MEASUREMENTS:
+                newMeasurement = CalculateMeanValue(storage_measurement_dict=storageMeasurementDict_1800s,
+                                                    measurement_key=measurementKey)
+                measurementsDict = CreateIndividualMeasurementDict(measurement_key=measurementKey,
+                                                                   measurement_series='3600s',
+                                                                   new_measurement=newMeasurement)
+
+                measurementsTemplateDict.update(measurementsDict)
+            jsonMeasurementList.append(measurementsTemplateDict)
+            storageMeasurementDict = CreateStorageMeasurementDict()
+            storageMeasurementDict_1800s = CreateStorageMeasurementDict()
+            time_count = 0
+
+        time_pointer += timedelta(seconds=600)
```

Comment:
Isn't time_pointer a functionvariable here ?

Reply:
I use time_pointer as a variable to loop from the start_time to end_time and after every loop, the time is increased by 600 seconds by the timedelta function. 
what do you mean about functionvariable?

Reply:
Its a variable within the function scope. So changing it just before terminating the function would change anything right?